src/ucp/wireup/wireup_cm.c

```diff
@@ -243,10 +244,186 @@ ucs_status_t ucp_ep_client_cm_connect_start(ucp_ep_h ucp_ep,
     return status;
 }
 
+static unsigned ucp_cm_server_conn_request_progress(void *arg)
+{
+    ucp_conn_request_h               conn_request = arg;
+    ucp_listener_h                   listener     = conn_request->listener;
+    ucp_worker_h                     worker       = listener->worker;
+    ucp_ep_h                         ep;
+    ucs_status_t                     status;
+
+    ucs_trace_func("listener %p, connect request %p", listener, conn_request);
+
+    if (listener->conn_cb) {
+        listener->conn_cb(conn_request, listener->arg);
+        return 1;
+    }
+
+    UCS_ASYNC_BLOCK(&worker->async);
+    status = ucp_ep_create_server_accept(worker, conn_request, &ep);
+    if (status != UCS_OK) {
+        ucs_warn("server endpoint creation with connect request %p failed, status %s",
+                  conn_request, ucs_status_string(status));
+    }
+    UCS_ASYNC_UNBLOCK(&worker->async);
+    ucs_free(conn_request);
+    return 1;
+}
+
 void ucp_cm_server_conn_request_cb(uct_listener_h listener, void *arg,
                                    const char *local_dev_name,
                                    uct_conn_request_h conn_request,
                                    const uct_cm_remote_data_t *remote_data)
+{
+    ucp_listener_h ucp_listener = arg;
+    uct_worker_cb_id_t prog_id  = UCS_CALLBACKQ_ID_NULL;
+    ucp_conn_request_h ucp_conn_request;
+    ucs_status_t status;
+
+    ucp_conn_request = ucs_malloc(ucs_offsetof(ucp_conn_request_t, sa_data) +
+                                  remote_data->conn_priv_data_length,
+                                  "ucp_conn_request_h");
+    if (ucp_conn_request == NULL) {
+        ucs_error("failed to allocate connect request, rejecting connection request %p on TL listener %p",
+                  conn_request, listener);
+        goto err_reject;
+    }
+
+    ucp_conn_request->remote_dev_addr = ucs_malloc(remote_data->dev_addr_length,
```

Comment:
can we unite this with the allocation in line 283?
i think remote_data->conn_priv_data_length should actually be of constant size..

Reply:
device addr is not a part of private data, it's part of UCT API, in case of rdmacm it's IB addr. Do we have a constant in UCT which defines max device addr size?

Reply:
no constant size, but we can allocate dynamic size

Reply:
Why need to overcomplicate memory layout for non-fast path?